src/solvers/dg_curved/containers_1d.jl

```diff
@@ -61,8 +61,13 @@ function initialize_neighbor_connectivity!(left_neighbors, mesh::CurvedMesh{1})
     left_neighbors[1, cell_x] = cell_x - 1
   end
 
-  # Periodic boundary
-  left_neighbors[1, 1] = size(mesh, 1)
+  if mesh.periodicity
+    # Periodic boundary
+    left_neighbors[1, 1] = size(mesh, 1)
+  else
+    # Use boundary conditions
+    left_neighbors[1, 1] = -1
+  end
```

Comment:
What about implementing a method `isperiodic(mesh::CurvedMesh)` and using this here (and in other places where `mesh.periodicity` is used)? We already have `isperiodic(t::AbstractTree)`, so this would fit nicely, I think.